app/src/main/AndroidManifest.xml

```diff
@@ -21,10 +21,42 @@
             android:theme="@style/Theme.PersonalizedAppIconAndroid"
             >
             <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+
+        <!-- Default icon -->
+        <activity-alias
+            android:name=".MainActivityAlias1"
+            android:targetActivity=".MainActivity"
+            android:label="@string/app_name"
+            android:icon="@mipmap/ic_launcher"
+            android:roundIcon="@mipmap/ic_launcher_round"
+            android:enabled="true"
+            android:exported="true"
+            android:theme="@style/Theme.PersonalizedAppIconAndroid"
+            >
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity-alias>
+
+        <activity-alias
+            android:name=".MainActivityAlias2"
+            android:targetActivity=".MainActivity"
```

Comment:
So, if by some miracle we always point to `MainActivityAlias1` as the default, when the app switches to `MainActivityAlias2` in order to change the app icon, what happens to the Android app link, then?